### PR TITLE
PT-1378 Represent pen bolus data type

### PR DESCRIPTION
**Pull request by franck-fourel**

### CODE REVIEW

File: docs/UpdatedData.md [hunk 1]
```diff
@@ -176,6 +176,23 @@ __Note #2__: the `"part":2` object is not mandatory. The system can decide to ca
...
 
+## Pen bolus
+
+A `pen` bolus is a normal bolus administered manually with insulin pen or syringe. The subType associated to this type of bolus is `pen`. It contains a single field compared to other boluses. 
+- `normal` is similar to what is defined in `normal` bolus. 
+
+```json
+{
+  "time": "2020-05-12T12:00:00.000Z",
+  "timezoneOffset": 120,
```
Comment (yvon-dblg) [inline docs/UpdatedData.md]:
```suggestion
  "timezone": "Europe/Paris",
```
Using timezone, which should be mandatory, in the examples is better than using the timezoneOffset.
The offset is currently add automatically anyway by platform. But the timezone is not.

File: data/types/bolus/pen/pen.go [hunk 2]
```diff
@@ -0,0 +1,63 @@
+package pen
+
+import (
+	"github.com/tidepool-org/platform/data"
+	"github.com/tidepool-org/platform/data/types/bolus"
+	"github.com/tidepool-org/platform/structure"
+)
+
+const (
+	SubType = "pen" // TODO: Rename Type to "bolus/normal"; remove SubType
```
Comment (yvon-dblg) [inline data/types/bolus/pen/pen.go]:
```suggestion
	SubType = "pen" // TODO: Rename Type to "bolus/pen"; remove SubType
```

File: data/types/bolus/pen/pen.go [hunk 3]
```diff
@@ -0,0 +1,63 @@
...
+	if !validator.HasMeta() {
+		validator = validator.WithMeta(n.Meta())
+	}
+
+	n.Bolus.Validate(validator)
+
+	if n.SubType != "" {
+		validator.String("subType", &n.SubType).EqualTo(SubType)
+	}
+
```
Comment (yvon-dblg) [inline data/types/bolus/pen/pen.go:48]:
```suggestion
```
Theses lines from "normal" bolus are not necessary.
If subType is not "pen" this code won't be executed anyway.

File: data/types/bolus/pen/pen.go [hunk 4]
```diff
@@ -0,0 +1,63 @@
...
+	"github.com/tidepool-org/platform/data"
+	"github.com/tidepool-org/platform/data/types/bolus"
+	"github.com/tidepool-org/platform/structure"
+)
+
+const (
+	SubType = "pen" // TODO: Rename Type to "bolus/normal"; remove SubType
+
+	NormalMaximum = 100.0
+	NormalMinimum = 0.0
```
Comment (yvon-dblg) [inline data/types/bolus/pen/pen.go:13]:
```suggestion
	normalMaximum = 100.0
	normalMinimum = 0.0
```
We do not need to export theses values.

File: data/types/bolus/pen/pen_test.go [hunk 5]
```diff
@@ -0,0 +1,159 @@
...
+			DescribeTable("validates the datum",
+				func(mutator func(datum *pen.Pen), expectedErrors ...error) {
+					datum := dataTypesBolusPenTest.NewPen()
+					mutator(datum)
+					dataTypesTest.ValidateWithExpectedOrigins(datum, structure.Origins(), expectedErrors...)
+				},
+				Entry("succeeds",
+					func(datum *pen.Pen) {},
+				),
+				Entry("type missing",
```
Comment (yvon-dblg) [inline data/types/bolus/pen/pen_test.go:65]: Not sure if it is necessary to test the presence of type/subType.
It is already done in the bolus_test & base_test.

**Review by yvon-dblg — CHANGES_REQUESTED** [review]
One typo & few remarks.

File: data/types/bolus/pen/pen.go (hunk 3 above)
Comment (franck-fourel) [reply data/types/bolus/pen/pen.go:48]: That's true the way we call it through the bolus factory. 
But nothing else in the code is preventing it to happen.

File: data/types/bolus/pen/pen.go (hunk 4 above)
Comment (franck-fourel) [reply data/types/bolus/pen/pen.go:13]: again, tests are missing

File: data/types/bolus/pen/pen_test.go (hunk 5 above)
Comment (franck-fourel) [reply data/types/bolus/pen/pen_test.go:65]: yep that makes the tests pretty verbose. I followed what was done in similar objects

**Review by yvon-dblg — APPROVED** [review]
LGTM
@CedricPerotto any comments ?

**Review by CedricPerotto — APPROVED** [review]

**Pull request merged**